### PR TITLE
Check if user has a permission to block putting X item on their head.

### PR DESCRIPTION
**Pull request by darbyjack**
A possible solution for #2550

### CODE REVIEW

File: Essentials/src/com/earth2me/essentials/commands/Commandhat.java [hunk 1]
```diff
@@ -35,6 +35,10 @@ protected void run(final Server server, final User user, final String commandLab
         } else {
             final ItemStack hand = user.getItemInHand();
             if (hand != null && hand.getType() != Material.AIR) {
+                if (user.isAuthorized("essentials.hat.exempt." + hand.getType().name())) {
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/commands/Commandhat.java]: You should `toLowerCase` the material name.

**Review by mdcfe — CHANGES_REQUESTED** [review]

File: Essentials/src/com/earth2me/essentials/commands/Commandhat.java (hunk 1 above)
Comment (darbyjack) [reply Essentials/src/com/earth2me/essentials/commands/Commandhat.java]: I had considered that at first but I didn't see that in the original issue. Just pushed a change to handle that.

File: Essentials/src/com/earth2me/essentials/commands/Commandhat.java [hunk 2]
```diff
@@ -30,11 +30,15 @@ protected void run(final Server server, final User user, final String commandLab
                 final ItemStack air = new ItemStack(Material.AIR);
                 inv.setHelmet(air);
                 InventoryWorkaround.addItems(user.getBase().getInventory(), head);
-                user.sendMessage(tl("hatRemoved"));
+                user.sendMessage(tl("hatRemosved"));
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/commands/Commandhat.java]: Typo

Comment (darbyjack) [reply Essentials/src/com/earth2me/essentials/commands/Commandhat.java]: Fixed!

**Review by mdcfe — APPROVED** [review]

**Review by Ichbinjoe — DISMISSED** [review]

File: Essentials/src/com/earth2me/essentials/commands/Commandhat.java [hunk 3]
```diff
@@ -35,6 +35,10 @@ protected void run(final Server server, final User user, final String commandLab
         } else {
             final ItemStack hand = user.getItemInHand();
             if (hand != null && hand.getType() != Material.AIR) {
+                if (user.isAuthorized("essentials.hat.exempt." + hand.getType().name().toLowerCase())) {
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/commands/Commandhat.java]: I had previously approved this, but "hat exempt" doesn't make a lot of sense. Could change it to `essentials.hat.prevent-type.<type>`? In addition, this check needs to be replicated in `EssentialsPlayerListener` where the `allow-direct-hat` handling happens.

**Review by mdcfe — CHANGES_REQUESTED** [review]

**Review by mdcfe — APPROVED** [review]

**Pull request merged**